### PR TITLE
Paydirekt

### PR DESCRIPTION
**Pull request by ahalberkamp**
This pr added Integration tests  for "paydirekt" to the paymentadapter. 
For creating web test properly, the html-driver  replaced by  the chromedriver.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ jdk:
   - oraclejdk8
 services:
   - docker
+addons:
+  chrome: stable
```
Comment (ahalberkamp) [inline .travis.yml]: Needed for running webtests on staging

File: build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ buildscript {
     dependencies {
         classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.4'
         classpath 'org.ajoberstar:gradle-git-publish:0.3.2'
-        classpath 'org.ajoberstar:grgit:2.1.0'
+        classpath 'org.ajoberstar:grgit:2.3.0'
```
Comment (ahalberkamp) [inline build.gradle]: needed for pushing test result to gh-pages

File: functionaltests/src/test/java/util/WebDriverPaydirekt.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * @author fhaertig
+ * @since 14.03.16
+ */
+public class WebDriverPaydirekt extends CustomWebDriver {
+
+    private static String LOGIN_FORM_USERNAME_FIELD_ID = "username";
+    private static String LOGIN_FORM_PASSWORD_FIELD_ID = "password";
+    private static String LOGIN_FORM_SUBMIT_BUTTON_NAME = "loginBtn";
+    private static String PAYMENT_SUBMIT_BUTTON_XPATH = "//BUTTON[@type='submit'][text()=' Jetzt bezahlen ']";
```
Comment (butenkor) [inline functionaltests/src/test/java/util/WebDriverPaydirekt.java]: Is it really the most reliable way to access the button?

File: functionaltests/src/test/java/util/CustomWebDriver.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            throw new RuntimeException("Submit button not found on the page");
+        }
+        return elements;
+    }
+
+    public ChromeDriver getDriver() {
+        return driver;
+    }
+
+    public void quit() {
```
Comment (butenkor) [inline functionaltests/src/test/java/util/CustomWebDriver.java]: Is it ever used? Same goes for all its subclasses.

Comment (ahalberkamp) [reply functionaltests/src/test/java/util/CustomWebDriver.java]: yes this is used. for example here:https://github.com/commercetools/commercetools-payone-integration/blob/paydirekt_it/functionaltests/src/test/java/util/WebDriverPaydirekt.java#L57

File: functionaltests/src/test/java/util/CustomWebDriver.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+     * Search for the single submit button on the page by className pattern from
+     * {@link util.constant.WebDriverCommon#CLASS_NAME_PRIMARY}
+     *
+     * @return found button.
+     * @throws RuntimeException if submit button can't be found or multiply buttons exist on the page
+     */
+    protected WebElement findSubmitButtonByName(String name) {
+        WebElement elements = findElement(By.name(name));
+        if (elements == null) {
+            throw new RuntimeException("Submit button not found on the page");
```
[the remark this answers is not in the document]
Comment (ahalberkamp) [reply functionaltests/src/test/java/util/CustomWebDriver.java]: ddf92830ee217ae605c2cc86b542930ca6331201

File: functionaltests/src/test/java/specs/paymentmethods/paydirekt/ChargeImmediatelyFixture.java [hunk 6]
```diff
@@ -27,23 +27,21 @@
...
+    private Map<String, String> successUrlForPayment;
 
     public boolean executeRedirectForPayments(final String paymentNames) throws ExecutionException {
         final Collection<String> paymentNamesList = ImmutableList.copyOf(thePaymentNamesSplitter.split(paymentNames));
 
-        // run all 3 payments approval in parallel, aka 3 different sessions
-        // and collect successfully approved redirect URLs
-        successUrlForPayment = paymentNamesList.stream().parallel()
+
+        successUrlForPayment = paymentNamesList.stream()
```
[the remark this answers is not in the document]
Comment (ahalberkamp) [reply functionaltests/src/test/java/specs/paymentmethods/paydirekt/ChargeImmediatelyFixture.java]: efa54e2b8b960b20b71b1e2bc4398dd3a893f2fa

File: functionaltests/src/test/java/specs/BaseFixture.java [hunk 7]
```diff
@@ -395,6 +397,14 @@ protected static String getTestDataSwBankTransferBic() {
         return getConfigurationParameter(TEST_DATA_SW_BANK_TRANSFER_BIC);
     }
 
+    public static String getTestDataPaydirectLogin() {
+        return getConfigurationParameter(TEST_DATA_PAYDIRECT_LOGIN);
```
[the remark this answers is not in the document]
Comment (ahalberkamp) [reply functionaltests/src/test/java/specs/BaseFixture.java]: fd9a7343d0de692cc590da250de19d7e058fc29f

File: functionaltests/src/test/java/util/WebDriver3ds.java [hunk 8]
```diff
@@ -55,17 +30,18 @@ public WebDriver3ds() {
...
         pwInput.sendKeys(password);
         final WebElement submitButton = findElement(By.cssSelector(CSS_SELECTOR_SEND));
         submitButton.click();
 
         // Wait for redirect to complete
-        final Wait<WebDriver> wait = new WebDriverWait(this, 10);
+        WebDriverWait wait = new WebDriverWait(getDriver(), 10);
         wait.until(not(ExpectedConditions.urlContains(URL_3DS_PATTERN)));
-        return getCurrentUrl();
+        Boolean waitToSuccess = wait.until(ExpectedConditions.urlContains("-Success"));
```
[the remark this answers is not in the document]
Comment (ahalberkamp) [reply functionaltests/src/test/java/util/WebDriver3ds.java]: ec892e7bc9bf0ee73e32fa0d4e4365e0f4bbe7e2

File: functionaltests/src/test/java/util/CustomWebDriver.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+
+public class CustomWebDriver {
+    private ChromeDriver driver;
+
+    CustomWebDriver() {
+        String currentDriver = SystemUtils.IS_OS_MAC ? "chromedriver_ios_2.4" : "chromedriver_unix_2.4";
```
[the remark this answers is not in the document]
Comment (ahalberkamp) [reply functionaltests/src/test/java/util/CustomWebDriver.java]: 00f97918

File: functionaltests/src/test/java/util/WebDriverPaydirekt.java (hunk 3 above)
Comment (butenkor) [reply functionaltests/src/test/java/util/WebDriverPaydirekt.java]: Thanks for extensive answers :) What other options did you try out?

File: functionaltests/src/test/java/util/CustomWebDriver.java (hunk 4 above)
Comment (butenkor) [reply functionaltests/src/test/java/util/CustomWebDriver.java]: I see that it only defined there like here but never really called, right?

Comment (ahalberkamp) [reply functionaltests/src/test/java/util/CustomWebDriver.java]: 38680277

File: functionaltests/src/test/java/util/WebDriverPaydirekt.java (hunk 3 above)
Comment (ahalberkamp) [reply functionaltests/src/test/java/util/WebDriverPaydirekt.java]: Now I locate the form, which I want to submit. Within this form, I locate the submit-button and then I click on it -> 0f6f0606

**Pull request merged**